src/main/java/io/orkes/conductor/client/automator/TaskRunner.java

```diff
@@ -175,7 +192,22 @@ private List<Task> pollTasksForWorker() {
             LOGGER.debug("Time taken to poll {} task with a batch size of {} is {} ms", taskType, tasks.size(), stopwatch.elapsed(TimeUnit.MILLISECONDS));
 
         }  catch (Throwable e) {
-            LOGGER.error("Error when polling for tasks: {}", e.getMessage(), e);
+            //For the first N (errorAt) errors, just print them as is...
+            boolean printError = false;
+            if(pollingErrorCount < errorAt) {
+                printError = true;
+            } else if (pollingErrorCount % errorAt == 0) {
+                printError = true;
+            }
+            pollingErrorCount++;
+            if(pollingErrorCount > 1_000_000) {
+                //Reset after 1 million errors
+                pollingErrorCount = 0;
+            }
+            if(printError) {
+                LOGGER.error("Error polling for taskType: {}, error = {}", taskType, e.getMessage(), e);
+            }
+            pollingErrorCount++;
```

Comment:
I don't understand this logic. What are we trying to do here?

Reply:
resetting this every million error back to printing - also early fencing around Integer overflow -- poor man's version. :)